pkg/build/controller/buildpod/factory.go

```diff
@@ -0,0 +1,213 @@
+package buildpod
```

Comment:
I'm not convinced separating that is necessary or even desirable.  I generally like it when all my object methods are in the same file as the object definition itself.  We don't create separate files for all our other New*() functions.